azure-iot-device/azure/iot/device/common/mqtt_transport.py

```diff
@@ -236,12 +238,20 @@ def connect(self, password=None):
         The password is not required if the transport was instantiated with an x509 certificate.
 
         :param str password: The password for connecting with the MQTT broker (Optional).
+
+        :raises: ConnectionFailedError if connection could not be established.
+        :raises: ConnectionDroppedError if connection is dropped during execution.
+        :raises: UnauthorizedError if there is an error authenticating.
+        :raises: ProtocolClientError if there is some other client error.
         """
         logger.info("connecting to mqtt broker")
 
         self._mqtt_client.username_pw_set(username=self._username, password=password)
 
-        rc = self._mqtt_client.connect(host=self._hostname, port=8883)
+        try:
+            rc = self._mqtt_client.connect(host=self._hostname, port=8883)
+        except Exception:
```

Comment:
>Exception [](start = 15, length = 9)

is it possible to pass the original exception while initializing the ProtocolClientError or is it not helpful ?